Makefile

```diff
@@ -0,0 +1,89 @@
+
+# NOTE: This is not the most perfect makefile.
+# There's a few things we could do to improve performance.
+# But since this is new, we can strive for clarity and simplicity here.
+# Could use many of these suggestions:
+# 	https://docs.cloudposse.com/reference/best-practices/make-best-practices/
+
+
+# On _some_ systems, make might not default to bash, which might produce
+# unexpected output/errors. It's assumed that all of the recipes here are written
+# in bash syntax.
+SHELL = /bin/bash
+.SHELLFLAGS = -e -u -o pipefail -c
+
+
+default: help
+
+
+# TODO: Add more descriptions to makefile targets
+.PHONY: help
+help:
+	@echo -e "\nThese are some useful commands to work with this project."
+	@echo -e "Please refer to the README.md for more details.\n"
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+	@echo -e "\nFor more information, please read the Makefile\n"
+
+
+.env:
+	@echo "Missing '.env' file. Creating one using 'env.example' as a template"
+	@cp env.example .env
+	@echo "Please open the '.env' file and adjust to your local setup!"
+
+
+.venv:  ## Creates a virtualenv at `./.venv`
+	@python -m venv .venv
+
+
+.PHONY: setup
+setup: .env .venv requirements.txt.log requirements-mysql.txt.log requirements-postgresql.txt.log  ## Sets up the project (installs dependencies etc.)
+
+
+requirements.txt.log: requirements.txt
+	@pip install -r requirements.txt | tee .requirements.txt.tmp.log
+	@mv .requirements.txt.tmp.log requirements.txt.log
```

Comment:
These `....log` targets allow us to cache installations of the requirements, which can speed up development.
Using these, the `setup` target above will be "cached" by make, so subsequent runs won't actually try and install already installed packages.